File: nim/armstrong-numbers/Iteration2/armstrong_numbers.nim
import math
  
proc isArmstrongNumber*(number: int): bool =
  var digits: seq[int]
  var numberOfDigits: int
  var current = number

  while current != 0:
    digits.add(current mod 10)
    numberOfDigits += 1
    current = current div 10

  var total: int

  for digit in digits:
    total += digit ^ numberOfDigits

  number == total
    

File: zig/armstrong-numbers/Iteration2/armstrong_numbers.zig
const std = @import("std");

pub fn isArmstrongNumber(num: u128) bool {
    if (num == 0) return true;
    
    const digits = std.math.log10_int(num) + 1;

    var n = num;
    var sum: u128 = 0;

    while (n > 0) : (n /= 10) {
        sum += std.math.pow(u128, n % 10, digits);
    }
    
    return num == sum;
}
